tests/test_config.py

```diff
@@ -1,71 +1,109 @@
-
-        old_value = config.instance_id
-        new_value = 'new'
-        config.instance_id = new_value
-        self.assertEqual(config.instance_id, new_value, 'New value was not set')
-        self.assertEqual(config['instance_id'], new_value, 'New value was not set')
-        self.assertEqual(config._instance.instance_id, old_value, 'Private value has changed')
-
-        config.reset('sh_base_url')
-        config.reset(['aws_access_key_id', 'aws_secret_access_key'])
-        self.assertEqual(config.instance_id, new_value, 'Instance ID should not reset yet')
-        config.reset()
-        self.assertEqual(config.instance_id, config._instance.CONFIG_PARAMS['instance_id'],
-                         'Instance ID should reset')
-
-    def test_save(self):
-        config = SHConfig()
-        old_value = config.download_timeout_seconds
-        config.download_timeout_seconds = "abcd"
-        self.assertRaises(ValueError, config.save)
-        new_value = 150.5
-        config.download_timeout_seconds = new_value
-        config.save()
-        config = SHConfig()
-        self.assertEqual(config.download_timeout_seconds, new_value, "Saved value has not changed")
-        config.download_timeout_seconds = old_value
+import pytest
+
+from sentinelhub import SHConfig
+
+
+def test_config_file():
+    config = SHConfig()
+
+    config_file = config.get_config_location()
+    assert os.path.isfile(config_file), f'Config file does not exist: {os.path.abspath(config_file)}'
+
+    with open(config_file, 'r') as fp:
+        config_dict = json.load(fp)
+
+    for param, value in config_dict.items():
+        if param in config._instance.CREDENTIALS:
+            continue
+
+        if isinstance(value, str):
+            value = value.rstrip('/')
+
+        assert config[param] == value
+
+
+def test_reset():
+    config = SHConfig()
+
+    old_value = config.instance_id
+    new_value = 'new'
+    config.instance_id = new_value
+    assert config.instance_id == new_value, 'New value was not set'
+    assert config['instance_id'] == new_value, 'New value was not set'
+    assert config._instance.instance_id == old_value, 'Private value has changed'
+
+    config.reset('sh_base_url')
+    config.reset(['aws_access_key_id', 'aws_secret_access_key'])
+    assert config.instance_id == new_value, 'Instance ID should not reset yet'
+
+    config.reset()
+    assert config.instance_id == config._instance.CONFIG_PARAMS['instance_id'], 'Instance ID should reset'
+
+
+def test_save():
+    config = SHConfig()
+    old_value = config.download_timeout_seconds
+
+    config.download_timeout_seconds = 'abcd'
+    with pytest.raises(ValueError):
         config.save()
 
-    def test_raise_for_missing_instance_id(self):
-        config = SHConfig()
+    new_value = 150.5
+    config.download_timeout_seconds = new_value
+    config.save()
+    config = SHConfig()
+    assert config.download_timeout_seconds == new_value, 'Saved value should have changed'
```

Comment:
Since this can fail, the *restore* in the following lines could not happen. One way to avoid this could be through using fixtures in the same way as when  testing ray clusters in eo-learn (so that the fixture restores the SHConfig after the test).

Reply:
I agree and added the fixture. I made sure the config is restored even if a test failed.

Reply:
Perfect!